File: HackerRank/day8/main.js
function processData(input) {
    //Enter your code here
    const lines = input.split('\n');
    const n = parseInt(lines[0]);
    
    const phoneBook = {};
    for (let i = 1; i <= n; i++) {
        const [name, number] = lines[i].split(' ');
        phoneBook[name] = number;
    }
    
    let output = [];
    for (let i = n + 1; i < lines.length; i++) {
        const name = lines[i].trim();
        if (name === '') continue;
        if (phoneBook.hasOwnProperty(name)) {
            output.push(`${name}=${phoneBook[name]}`);
        } else {
            output.push('Not found');
        }
    }
    
    console.log(output.join('\n'));
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: HackerRank/day8/main.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        Dictionary<string, string> phoneBook = new Dictionary<string, string>();
        for (int i = 0; i < n; i++) {
            string[] entry = Console.ReadLine().Split(' ');
            phoneBook[entry[0]] = entry[1];
        }

        string name;
        while ((name = Console.ReadLine()) != null) {
            name = name.Trim();
            if (name == "") continue;
            if (phoneBook.ContainsKey(name)) {
                Console.WriteLine($"{name}={phoneBook[name]}");
            } else {
                Console.WriteLine("Not found");
            }
        }
    }
}